### ISSUE TITLE
Fidelity gate: MakeConsumerWithTwoLeadingArgs has a new raised opcode diff (spurious stloc/ldloc)

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3490]
## Summary

`FidelityGateTests.NoNewFidelityDiffsBeyondKnownDocket` fails:
`CfgSampleClass.MakeConsumerWithTwoLeadingArgs` produces a new `OpcodeDiff` that
is not in the raised gate's `KnownDiffs` docket. The recompiled body round-trips
the object-initializer receiver through a local instead of keeping it on the
stack.

## Evidence

```
MakeConsumerWithTwoLeadingArgs [OpcodeDiff]
  original  : ldarg call ldarg call newobj dup ldarg callvirt dup ldarg callvirt newobj ret
  recompiled: ldarg call ldarg call stloc ldloc newobj dup ldarg callvirt dup ldarg callvirt newobj ret
```

The inserted `stloc`/`ldloc` pair is the whole delta; opcode families otherwise
match in order.

Fixture: `src/ILInspector.Decompiler.Tests/CfgSampleClass.cs:2981`

## Note on scope

This method is *already* a documented known diff in the **lowered** gate
(`LoweredFidelityGateTests.cs:161`, attributed to #3272, "the trailing object
initializer ..."), but it is not in the **raised** gate's docket. So either the
raised path has regressed to match a shape only the lowered path was expected to
show, or the raised docket needs the same entry with a rationale. Worth deciding
which before pinning it.

## Why this was not caught

Class-level `[Trait("Speed", "Slow")]` (`FidelityGateTests.cs:14`); every
pre-merge lane runs `-trait- "Speed=Slow"` (`.github/workflows/ci.yml:301`). The
lanes that do include it (`release.yml`, weekly Deep Inspect) were exceeding
their job timeout, and Deep Inspect's failure notifier is gated on `failure()`,
which a *cancelled* job does not satisfy. See #3432.

## Repro

```bash
dotnet build dotnet-inspect.slnx -c Release
dotnet run --project src/ILInspector.Decompiler.Tests -c Release -- \
  -class ILInspector.Decompiler.Tests.FidelityGateTests
```

Observed on CI run 30440377772 (ubuntu-26.04, Release).

**Comment by richlander** [comment]
## Triage at `e1b3cac9`: the raised gate now has **three** new diffs, not one

Reproduced on a clean worktree detached at main's tip (`e1b3cac9`, #3495's merge), `dotnet build dotnet-inspect.slnx -c Release`, then:

```bash
dotnet run --project src/ILInspector.Decompiler.Tests -c Release --no-build -- \
  -class ILInspector.Decompiler.Tests.FidelityGateTests
```

`MakeConsumerWithTwoLeadingArgs` reproduces exactly as filed. But `NoNewFidelityDiffsBeyondKnownDocket` reports two further entries that are not in this issue and were not in CI run 30440377772:

```
ManualConstantOnlyComparisonFactory [OpcodeDiff]
  original  : ldtoken call ldstr call stloc ldc.i4 box ldtoken call call ldc.i4 box
              ldtoken call call call ldc.i4 newarr dup ldc.i4 ldloc stelem.ref call ret
  recompiled: ldtoken call ldstr call stloc ldc.i4 box ldtoken call call ldc.i4 box
              ldtoken call call call stloc ldc.i4 newarr dup ldc.i4 ldloc stelem.ref
              stloc ldloc ldloc call ret

StatementBodyLambdaInsideIf [OperandDiff]
  original  : ldarg brfalse ldsfld dup brtrue pop ldsfld ldftn newobj dup stsfld ret ldnull ret
  recompiled: ldarg brfalse ldsfld dup brtrue pop ldsfld ldftn newobj dup stsfld ret ldnull ret
  fidelity  : Removed ldsfld ...CfgSampleClass+<>c::<>9__103_0
              Added   ldsfld ...CfgSampleClass+<>c::<>9__128_0
              Removed ldftn  ...CfgSampleClass+<>c::<StatementBodyLambdaInsideIf>b__103_0(int32)
              Added   ldftn  ...CfgSampleClass+<>c::<StatementBodyLambdaInsideIf>b__128_0(int32)
              Removed stsfld ...CfgSampleClass+<>c::<>9__103_0
              Added   stsfld ...CfgSampleClass+<>c::<>9__128_0
```

### These two are already-understood families, and neither is a new decompiler defect

**`StatementBodyLambdaInsideIf` is #3491's root cause in a second spelling.** The opcode streams are character-for-character identical; the only delta is the Roslyn mangling ordinal, `103` → `128`. That is the enclosing method's positional index in its type, which compile-back cannot preserve because it reconstructs `CfgSampleClass` in a different member order. See the traced comment on #3491.

This matters for scoping that fix. #3491 proposes an `IlBodyDiffNormalization` flag folding `|N_K` to `|*_K` — the **local-function** spelling. This entry is the **lambda-cache** spelling, `<>9__N_K` and `<Enclosing>b__N_K`. A fold written only for `|N_K` will not retire it, and this gate will still be red.

**It also confirms #3491's drift prediction empirically.** This method passed in CI run 30440377772 and fails now, roughly three hours later, with no decompiler change in between — only unrelated merges into `main`. #3491 says pass/fail is coincidental because both ordinals move whenever anyone adds a method to the fixture file. This is that happening, observed live. It makes the normalization the highest-value of the five open regressions: it is the only one that keeps re-breaking on unrelated traffic.

**`ManualConstantOnlyComparisonFactory` is the stack-slot/temporary reshaping family**, whose siblings are already docketed in the lowered gate (`LoweredFidelityGateTests.cs:106-111,119-121`) — and it is simultaneously one of #3491's six lowered entries. The inserted `stloc`/`ldloc` pairs are the same shape.

### Which leaves the actual question this issue asks

`MakeConsumerWithTwoLeadingArgs` is docketed in the **lowered** gate (`LoweredFidelityGateTests.cs:161`, attributed to #3272) but not the **raised** one, and its delta here is a single `stloc`/`ldloc` pair around the object-initializer receiver — the same shape as its lowered rationale. So the decision this issue flags is still the right one to make, and it now applies to two methods rather than one: either the raised path regressed into a lowered-only shape, or the raised docket needs the same entries with the same rationale.

Recommend #3491's normalization lands first and is scoped to both mangling spellings. That removes `StatementBodyLambdaInsideIf` from this list and stops the gate flapping, leaving a clean two-method decision here.

**Comment by richlander** [comment]
## Verified fixed at `f3bc7b09` — closing

#3528 lists this issue under `Fixes`, but GitHub only auto-closes the first reference in a comma-separated list, so it stayed open. Verified directly rather than taking the PR's word for it.

Clean worktree at `origin/main` (`f3bc7b09`), `dotnet build dotnet-inspect.slnx -c Release` clean, then the full class with its `Speed=Slow` gates included (no `-trait-` filter):

```
FidelityGateTests                Total: 6   Failed: 0   182.4s
LoweredFidelityGateTests         Total: 5   Failed: 0   177.1s
ValidityCoverageReportingTests   Total: 3   Failed: 0    16.5s
GeneratedFixtureCatalogTests     Total: 18  Failed: 0    33.5s
```

The runtimes confirm the slow gates actually executed rather than being filtered out — these are the same classes that took 210s / 200s / 28s / 45s when they were failing during triage at `e1b3cac9`.

All four classes that carried the five regressions are green. The three additional raised-gate diffs I reported during triage — `ManualConstantOnlyComparisonFactory` and `StatementBodyLambdaInsideIf` on top of the filed `MakeConsumerWithTwoLeadingArgs` — are also gone.
